src/appkit/app/delegate.rs

```diff
@@ -307,150 +307,141 @@ pub(crate) fn register_app_delegate_class<T: AppDelegate + AppDelegate>() -> *co
+        decl.add_method(sel!(applicationWillUpdate:), will_update::<T> as extern "C" fn(_, _, _));
+        decl.add_method(sel!(applicationDidUpdate:), did_update::<T> as extern "C" fn(_, _, _));
         decl.add_method(
             sel!(applicationShouldHandleReopen:hasVisibleWindows:),
-            should_handle_reopen::<T> as extern "C" fn(&Object, _, _, BOOL) -> BOOL
+            should_handle_reopen::<T> as extern "C" fn(_, _, _, _) -> _
         );
 
         // Dock Menu
-        decl.add_method(
-            sel!(applicationDockMenu:),
-            dock_menu::<T> as extern "C" fn(&Object, _, _) -> id
-        );
+        decl.add_method(sel!(applicationDockMenu:), dock_menu::<T> as extern "C" fn(_, _, _) -> _);
 
         // Displaying Errors
         decl.add_method(
             sel!(application:willPresentError:),
-            will_present_error::<T> as extern "C" fn(&Object, _, _, id) -> id
+            will_present_error::<T> as extern "C" fn(_, _, _, _) -> _
         );
 
         // Managing the Screen
         decl.add_method(
             sel!(applicationDidChangeScreenParameters:),
-            did_change_screen_parameters::<T> as extern "C" fn(&Object, _, _)
+            did_change_screen_parameters::<T> as extern "C" fn(_, _, _)
         );
         decl.add_method(
             sel!(applicationDidChangeOcclusionState:),
-            did_change_occlusion_state::<T> as extern "C" fn(&Object, _, _)
+            did_change_occlusion_state::<T> as extern "C" fn(_, _, _)
         );
 
         // User Activities
         decl.add_method(
             sel!(application:willContinueUserActivityWithType:),
-            will_continue_user_activity_with_type::<T> as extern "C" fn(&Object, _, _, id) -> BOOL
+            will_continue_user_activity_with_type::<T> as extern "C" fn(_, _, _, _) -> _
         );
         decl.add_method(
             sel!(application:continueUserActivity:restorationHandler:),
-            continue_user_activity::<T> as extern "C" fn(&Object, _, _, id, id) -> BOOL
+            continue_user_activity::<T> as extern "C" fn(_, _, _, _, _) -> _
         );
         decl.add_method(
             sel!(application:didFailToContinueUserActivityWithType:error:),
-            failed_to_continue_user_activity::<T> as extern "C" fn(&Object, _, _, id, id)
+            failed_to_continue_user_activity::<T> as extern "C" fn(_, _, _, _, _)
         );
         decl.add_method(
             sel!(application:didUpdateUserActivity:),
-            did_update_user_activity::<T> as extern "C" fn(&Object, _, _, id)
+            did_update_user_activity::<T> as extern "C" fn(_, _, _, _)
         );
 
         // Handling push notifications
         decl.add_method(
             sel!(application:didRegisterForRemoteNotificationsWithDeviceToken:),
-            registered_for_remote_notifications::<T> as extern "C" fn(&Object, _, _, id)
+            registered_for_remote_notifications::<T> as extern "C" fn(_, _, _, _)
```

Comment:
Am I correct in that with your changes/`objc2`, we don't need to specify the method argument types anymore - i.e, the freestanding function can simply configure them as parameters?

Reply:
Yes, this is a change in `objc2`, see https://github.com/madsmtm/objc2/pull/193 and the links in there for details (sorry, the issue is a bit difficult to explain, and I probably don't really understand it fully myself).

Note that this is not just a convenience change, you are _required_ to use `_` now since there is no syntax to specify the correct lifetime `'a` in `fn(&'a Object, Sel, ...)`.

The upside is that things are more consistent, for example you can now use `&Object` in arguments as well, and you no longer need to specify the types twice.